tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -298,7 +299,7 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
         {
             if (parent == null && !ignoreActiveScope)
```

Comment:
I'm worried about the scenario where an application uses `StartActive(operationName, parent)` where `parent` is a `SpanContext` from the manual tracing assembly. Is that being tested in this PR? And how might we handle that? The scenarios where that could happen right now is where a customer has to manually link some sqs/kafka messages to local traces

Reply:
I didn't think of testing this scenario. I don't actually know what's the expected behavior in that scenario.

It's pretty much the same thing as if the user was doing:
```csharp
using (var parent = Tracer.StartActive("Parent"))
{
    using (var child = Tracer.StartActive("Child", spanContextFromSomewhereElse))
    {
          // What does child look like?
    } 
}
```

I suppose that's what the `ignoreActiveScope` parameter is there for?

Reply:
Thinking about this more, I think how we handle this is actually fine. The `ScopeManager` and `Scope` objects are more-or-less a glorified stack of `Span` objects whereas the `SpanContext` and `TraceContext` explain the relationships to the Span/Trace. If we don't connect the new Span to the 'Active' one or we choose to be a child of a different 'Span' then the only consequence is that the `SpanContext` and `TraceContext` may hold different values, and the logic in this PR essentially makes sure the information about the 'active' Span and its trace is kept up-to-date by both manual and automatic sides. So, I think we're good here